### PR TITLE
feat: add variance function

### PR DESCRIPTION
**Pull request by vibhatha**
This PR includes the addition of variance aggregation operator.

### CODE REVIEW

**Review by jvanstraten — CHANGES_REQUESTED** [review]
Same as for #257: not sure how to feel about the integer variants, and the floating point operations should use float rounding options rather than integer overflow options.

**Comment by vibhatha** [comment]
> Same as for #257: not sure how to feel about the integer variants, and the floating point operations should use float rounding options rather than integer overflow options.

@jvanstraten I understand. We can hold until the decision is finalized.

**Comment by jvanstraten** [comment]
YAML CI:

```
functions_arithmetic.yaml invalid
[
  {
    instancePath: '/aggregate_functions/4/impls/0',
    schemaPath: '#/properties/impls/items/required',
    keyword: 'required',
    params: { missingProperty: 'intermediate' },
    message: "must have required property 'intermediate'"
  }
]
```

@jacques-n I forgot which earlier PR this came up in. Could that have been one of the PRs you closed yesterday, maybe? I figured it'd be in main already.

**Comment by vibhatha** [comment]
@jvanstraten I updated the PR.

File: extensions/functions_arithmetic.yaml [hunk 1]
```diff
@@ -573,6 +573,23 @@ aggregate_functions:
...
+            required: false
+          - value: fp32
+        nullability: DECLARED_OUTPUT
+        return: fp32?
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: fp64
+        intermediate: "fp64?"
```
Comment (jvanstraten) [inline extensions/functions_arithmetic.yaml:591]: Looks like you might have deleted the wrong line here:

```suggestion
        nullability: DECLARED_OUTPUT
```

**Review by jvanstraten — CHANGES_REQUESTED** [review]
LGTM aside from the above.

**Comment by vibhatha** [comment]
@jvanstraten updated with the suggestion.

**Comment by jvanstraten** [comment]
Same CI issues as #257, but LGTM otherwise.

**Comment by vibhatha** [comment]
@jvanstraten updated the PR, there was a typo. Let's check the CIs once again. Sorry about this.

**Comment by vibhatha** [comment]
@jvanstraten I squashed the commits to a single one. Let's see the CIs.

**Review by jvanstraten — APPROVED** [review]

**Comment by jacques-n** [comment]
Hey @vibhatha , looks like other merges caused a merge conflict here. Can you rebase/fix?

Thanks!

**Comment by vibhatha** [comment]
@jacques-n shall we go with this: https://github.com/substrait-io/substrait/pull/257 first and I will resolve conflicts after this.

**Comment by jacques-n** [comment]
#257 is merged now.

**Comment by vibhatha** [comment]
@jacques-n could you please check the content after rebase.

**Pull request merged**